src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Uap.cs

```diff
@@ -26,13 +27,24 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
             // but we can at least get the path to the executable file for the process, and
             // other than for debugging tools, that's the main reason consumers of Modules care about it,
             // and why MainModule exists.
+            char[] chars = null;
+            int capacity = 64, length = 0;
             try
             {
                 // Get the path to the executable
                 using (Process process = Process.GetProcessById(processId))
                 {
-                    char[] chars = new char[1024];
-                    int length = Interop.Kernel32.GetModuleFileNameEx(process.SafeHandle, IntPtr.Zero, chars, chars.Length);
+                    do
+                    {
+                        if(chars != null)
+                        {
+                            ArrayPool<char>.Shared.Return(chars);
```

Comment:
No, if you're going to need another immediately. Just return it when you're done with this loop, or the function.